lib/rendition.js

```diff
@@ -41,12 +74,74 @@ class Rendition {
         }
     }
 
+    exists() {
+        return fs.existsSync(this.path);
+    }
+
     size() {
-        return this.metadata[RenditionMetadata.REPO_SIZE];
+        return fs.statSync(this.path).size;
     }
 
-    sha1() {
-        return this.metadata[RenditionMetadata.REPO_SHA1];
+    async sha1() {
+        if (!this._sha1) {
```

Comment:
Nitpick - but I would add a check verifying the `this._sha1` is not blank (since nothing is really private in javascript, something could have changed it to a blank string).

By blank I mean containing chars like `\n` or `\t` only.

Reply:
Why? This is an internal variable (`_`) to cache the value of the potentially very expensive sha calculation of large files, and it needs to only run if not set yet.

If a code change would accidentally modify it, then it should not be hidden and lead to another run.... and such a modification could always unset it as well (e.g. `_sha1 = false`), which we cannot protect from in the condition here.